### DISCUSSION TITLE
Spawning/building entities incrementally (from rows in a CSV file)

### DISCUSSION THREAD

**Discussion by biot023** [bevyengine/bevy#5436] [Q&A] [answered]
Hello -- I want to specify all my possible entities in a CSV file.
I've got the code working, for now, but the idea is that the components that my spawning system will have to handle will become much more complex over time -- currently it's a manageable `if / else if / else` but that will become less usable rather quickly!
Is there any way to incrementally add components to an entity builder and then have it finish? That would make my life miles easier but I haven't been able to find a way to do it, yet.
Thanks,
   Doug.

**Comment by NiklasEi** [comment, marked as answer]
You could see the entity commands returned by `commands.spawn()` as an entity builder. I am on my phone, so no guarantee that it compiles, but it could look like this:
```
let mut entity = commands.spawn();
for component in components {
    entity.insert(component);
}
```

**Reply by biot023** [reply]
Cheers, @NiklasEi -- that doesn't work, though, as it complains that you're trying to use a reference beyond its lifetime.
Effectively, to take the `spawn()` approach, I've had to write conditional blocks of the same code minus or plus certain components, depending on the data in the entity template that came from the CSV file.
I'm sticking with that for the moment, at least until someone can suggest a better way.
Two possibilities would be:

- [bevy_proto](https://github.com/MrGVSV/bevy_proto) -- doesn't *exactly* do what I want but might become preferable as my approach becomes more-and-more unwieldy
- Something with a custom factory around implementing the `Bundle` trait -- although there's `unsafe` code and everything, in there, and I'm not overly keen on going down a rabbit-hole quite so clearly labelled with "Here be dragons"! :)

**Reply by NiklasEi** [reply]
Can I see the code somewhere? The only thing that might get lifetime issues here is the components collection I think. There should be a nicer solution than duplication using just the entity commands.

**Comment by biot023** [comment]
Hello, again, and thank you.
I re-started the project to better structure it and, upon re-writing my `entity-to-spawn` system, I find it does indeed work exactly as you outline.
I'm not sure what I was getting wrong, sorry -- I'm still very much finding my feet.
Thanks so much for taking the time to answer my question.
   Doug.
